### DISCUSSION TITLE
How to control authorized users within the same connections?

### DISCUSSION THREAD

**Discussion by TelDragon** [strongswan/strongswan#2242] [Q&A]
connection demo:
```
connections {
    eap-1{
        version = 2
        proposals = default
        unique = never
        pools = Private_Network_1
        local {
            auth = pubkey
            certs = control_ServerCert.pem
            id = control.ipsec.xxx.com.cn
        }
        remote {
            auth = eap-mschapv2
            eap_id = %any
        }
        children {
            eap-1{
                start_action = start
                dpd_action = restart
                local_ts  = dynamic
                remote_ts = dynamic
                esp_proposals = default
            }
        }
    }
}
```

secrets demo
```
secrets {
   eap-control_1 {
      id = control_1
      secret = 123456
   }
   eap-control_2 {
      id = control_2
      secret = 123456
   }
  eap-control_3 {
      id = control_3
      secret = 123456
   }

.。。。。。。。
}
```

The main problem lies in the remote node. Exclude Windows end users.
On identifiable clients such as Android and Linux
I don't want to create multiple connections to implement. But EAP-1 specifies authorizing a certain ID or user to connect. For example, controll_1 and controll_2. Control_3 rejects the connection.

**Comment by tobiasbrunner** [comment]
I suggest looking into [group selection](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_group_selection) via RADIUS. Another option could be the [ext-auth plugin](https://docs.strongswan.org/docs/5.9/plugins/ext-auth.html).

**Comment by TelDragon** [comment]
> I suggest looking into [group selection](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_group_selection) via RADIUS. Another option could be the [ext-auth plugin](https://docs.strongswan.org/docs/5.9/plugins/ext-auth.html).

Thank you very much for your reply.
The architecture of RADIUS is a bit heavy for me. I am not very inclined towards this plan.
The function I want to implement is actually very simple, which is to restrict or specify multiple users through a single connection. Or client.
The use of ext auth plugin. Is there a demo reference in conf?

 [ext-auth plugin](https://docs.strongswan.org/docs/5.9/plugins/ext-auth.html) Here is only a brief explanation
